packages/astro/src/content/content-layer.ts

```diff
@@ -284,6 +284,13 @@ export async function simpleLoader<TData extends { id: string }>(
 	}
 }
 
+export function getDataStoreFile(settings: AstroSettings) {
+	return new URL(
+		DATA_STORE_FILE,
+		process?.env.NODE_ENV === 'development' ? settings.dotAstroDir : settings.config.cacheDir,
+	);
+}
```

Comment:
It would be nice to have an explanation why we pick different dirs for dev and build.